core/tests/webdriverio_utils/LearnerDashboardPage.js

```diff
@@ -0,0 +1,265 @@
+    await waitFor.textToBePresentInElement(
+      currentGoalsTopicName, name,
+      `Text "${name}" taking too long to be present in currentGoalsTopic`);
+    var topicName = $(
+      `.e2e-test-topic-name-in-current-goals=${name}`);
+    expect(await action.getText('Topic Name', topicName)).toMatch(name);
+  };
+
+  this.expectNameOfTopicInSkillProficiencyToMatch = async function(name) {
+    await waitFor.visibilityOf(
+      skillProficiencyTopicTitle,
+      'Topic in Skill Proficiency takes too long to appear'
+    );
+    await waitFor.textToBePresentInElement(
+      skillProficiencyTopicTitle, name,
+      `Text "${name}" taking too long to be present in skillProficiencyTopic`);
+    var topicName = $(
+      `.e2e-test-skill-proficiency-topic-title=${name}`);
+    expect(await action.getText('Topic Name', topicName)).toMatch(name);
+  };
+
+  this.expectNameOfTopicInCompletedGoalsToMatch = async function(name) {
+    await waitFor.visibilityOf(
+      completedGoalsTopicName,
+      'Topic in completed goals takes too long to appear'
+    );
+    await waitFor.textToBePresentInElement(
+      completedGoalsTopicName, name,
+      `Text "${name}" taking too long to be present in completedGoalsTopic`);
+    var topicName = $(
+      `.e2e-test-completed-goals-topic-name*=${name}`);
+    expect(await action.getText('Topic Name', topicName)).toMatch(name);
+  };
+
+  this.expectNumberOfTopicsInSuggestedForYou = async function(value) {
+    var topicNamesInLearnerTopicSummaryTiles = await $$(
+      '.e2e-test-learner-topic-summary-tile-title');
+    if (value > 0) {
+      await waitFor.visibilityOf(
+        topicNamesInLearnerTopicSummaryTiles[0],
+        'Learner Topic Name takes too long to appear');
+    }
+    expect(topicNamesInLearnerTopicSummaryTiles.length).toEqual(value);
+  };
+
+  this.expectNumberOfStoriesInCompletedStory = async function(value) {
+    var storyNamesInLearnerStorySummaryTiles = await $$(
+      '.e2e-test-story-name-in-learner-story-summary-tile');
+    if (value > 0) {
+      await waitFor.visibilityOf(
+        storyNamesInLearnerStorySummaryTiles[0],
+        'Story Name Card takes too long to appear');
+    }
+    expect(storyNamesInLearnerStorySummaryTiles.length).toEqual(value);
+  };
+
+  this.expectNumberOfTopicsInContinueWhereYouLeftOff = async function(value) {
+    var topicNamesInLearnerStorySummaryTiles = await $$(
+      '.e2e-test-topic-name-in-learner-story-summary-tile');
+    if (value > 0) {
+      await waitFor.visibilityOf(
+        topicNamesInLearnerStorySummaryTiles[0],
+        'Topic Name Card takes too long to appear');
+    }
+    expect(topicNamesInLearnerStorySummaryTiles.length).toEqual(value);
+  };
+
+  this.addTopicToLearnerGoals = async function() {
+    await action.click('Add to learner goals button', addToLearnerGoalsButton);
+  };
+
+  this.expectSubscriptionFirstNameToMatch = async function(name) {
+    await subscriptionSection.scrollIntoView();
```

Comment:
Why?

Reply:
getText in webdriverio only works if its visible else it returns an empty string. For more info PTAL [here](https://webdriver.io/docs/api/element/getText/)
Thanks